### PR TITLE
fix  'Calculator' page reverting after changing locale 

### PR DESCRIPTION
**Pull request by NVMakarenko**
dev
## PROJECT

[Board ticket](https://github.com/orgs/ita-social-projects/projects/25/views/7?pane=issue&itemId=59388724&issue=ita-social-projects%7CZeroWaste%7C866)

## Code reviewers

- [x] @loqimean 

## Summary of issue

New calculator is not creating because of validation failed. Error message is shown. If change locale on this stage, admin will be redirected to Calculator's page.
**Expected result**
'New calculator' page is reload and displayed.
'Name' field is blank.
Error message is absent.

## Summary of change

1. fix turbo_stream response.
2. move form into partial.

https://github.com/user-attachments/assets/fab3cc80-1e05-4f55-a19a-fb87c3592e38

## CHECK LIST
- [x]  СI passed
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in user
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/views/account/calculators/new.html.erb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <div class="container">
   <%= simple_form_for @calculator, url: account_calculators_path do |f| %>
+    <%= hidden_field_tag :rendered_from, "new" %>
```
Comment (loqimean) [inline app/views/account/calculators/new.html.erb:3]: не гуд, таких форм у тебе рахуй 10 і на кожну потрібна перевірка?) не гуд, шукай інший солюшен, більш лайтовий

**Review by loqimean — CHANGES_REQUESTED** [review]

**Review by KseniaGovorun — APPROVED** [review]

**Review by KseniaGovorun — CHANGES_REQUESTED** [review]
Fix ci

**Review by KseniaGovorun — APPROVED** [review]

File: app/views/account/calculators/create.turbo_stream.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<div data-controller="redirect" data-url="<%= account_calculators_path %>"></div>
```
Comment (loqimean) [inline app/views/account/calculators/create.turbo_stream.erb:1]:
```ruby
turbo_stream.append :body_js do
  <script>
    window.location.replace("<%= account_calculators_path %>")
  </script>
end
```

File: app/views/account/calculators/new.turbo_stream.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<%= turbo_stream.replace(dom_id(@calculator, :form), partial: "account/calculators/partials/new/form", locals: {calculator: @calculator}) %>
```
Comment (loqimean) [inline app/views/account/calculators/new.turbo_stream.erb:1]:
```suggestion
<%= turbo_stream.replace(dom_id(@calculator, :form), partial: "account/calculators/partials/new/form", locals: { calculator: @calculator }) %>

```

File: app/javascript/controllers/redirect_controller.js
Comment (loqimean) [inline app/javascript/controllers/redirect_controller.js:1]: а це можна видаляти

**Review by loqimean — CHANGES_REQUESTED** [review]

**Review by loqimean — APPROVED** [review]

**Pull request merged**